Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -13,18 +13,63 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string[] postTimes, string[] showTimes, string expected)
-        {
+        private static void Test(string[] postTimes, string[] showTimes, string expected)
+        {
             var result = GetCurrentTime(postTimes, showTimes).Equals(expected) ? "PASS" : "FAIL";
             var postTimesCsv = string.Join(", ", postTimes);
             var showTimesCsv = string.Join(", ", showTimes);
-            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
         }
 
-        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+        {
+            var length = exactPostTime.Length;
+            List<string> array = new List<string>();
+
+            for(var i=0;i<length;i++){
+                // Check whether for same exactPostTime we have different showPostTime
+
+                if(i!=length-1){
+                    if(exactPostTime[i]==exactPostTime[i+1] && showPostTime[i]!=showPostTime[i+1])
+                        return "impossible";
+                }
+
+                 var postTime = TimeSpan.Parse(exactPostTime[i]);
+
+                // add seconds in exactPostTime
+                if(showPostTime[i].Contains("seconds")){
+                    var currentTime = new TimeSpan(postTime.Hours,postTime.Minutes,postTime.Seconds);
+                    array.Add(Convert.ToString(currentTime));
+                     
+                }
+
+                // else add minutes in exactPostTime
+                else if(showPostTime[i].Contains("minutes")){
+                    var minutes = showPostTime[i].Split(" ")[0];
+                    postTime = postTime.Add(TimeSpan.FromMinutes(Double.Parse(minutes)));
+                    var currentTime = new TimeSpan(postTime.Hours,postTime.Minutes,postTime.Seconds);
+                    array.Add(Convert.ToString(currentTime));
+                }
+
+                // else add hours in exactPostTime
+                else if(showPostTime[i].Contains("hours")){
+                    var hours = showPostTime[i].Split(" ")[0];
+                    postTime = postTime.Add(TimeSpan.FromHours(Double.Parse(hours)));
+                    var currentTime = new TimeSpan(postTime.Hours,postTime.Minutes,postTime.Seconds);
+                    array.Add(Convert.ToString(currentTime));
+                }
+                else{
+                    return "impossible";
+                }
+
+            }
+
+            // Sort array and return the largest one
+
+            array.Sort();
+            return Convert.ToString(array[array.Count-1]);

```

Comment:
array is already list of string, so its element is of type string only.. You don't need to convert it to string again. It's just adding extra overhead to program by Converting string to string